src/Workspaces/Core/Portable/Workspace/Solution/Solution.CompilationTracker.cs

```diff
@@ -645,7 +645,7 @@ public CompilationInfo(Compilation compilation, bool hasCompleteReferences)
                         }
                     }
 
-                    if (!Enumerable.SequenceEqual(compilation.References, newReferences))
+                    if (!Enumerable.SequenceEqual(compilation.References, newReferences, MetadataReferenceComparer.Instance))
```

Comment:
Does this actually improve anything? As long as compilation.References includes at least one metadata reference your comparer will return false, and SequenceEqual will return true. If anything, since we're bypassing the PE reference comparer, I'd expect this makes things _worse_.